src/Command/SubProfileCommand.php

```diff
@@ -0,0 +1,363 @@
+      ->setHelp($this->trans('The <info>lightning:subprofile</info> command helps you generate a new subprofile of Lightning'))
+      ->addOption(
+        'profile',
+        '',
+        InputOption::VALUE_REQUIRED,
+        $this->trans('The name of the subprofile')
+      )
+      ->addOption(
+        'machine-name',
+        '',
+        InputOption::VALUE_REQUIRED,
+        $this->trans('The machine name of the subprofile (lowercase and underscore only)')
+      )
+      ->addOption(
+        'description',
+        '',
+        InputOption::VALUE_OPTIONAL,
+        $this->trans('Subprofile description')
+      )
+      ->addOption(
+        'core',
+        '',
+        InputOption::VALUE_OPTIONAL,
+        $this->trans('commands.generate.profile.options.core'),
+        '8.x'
+      )
+      ->addOption(
+        'dependencies',
+        false,
+        InputOption::VALUE_OPTIONAL,
+        $this->trans('commands.generate.profile.options.dependencies')
+      )
+      ->addOption(
+        'excluded_dependencies',
+        false,
+        InputOption::VALUE_OPTIONAL,
+        $this->trans('Top-level components of Lightning to exclude separated by commas.')
+      )
+      ->addOption(
+        'excluded_subcomponents',
+        false,
+        InputOption::VALUE_OPTIONAL,
+        $this->trans('Lightning sub-components to exclude separated by commas.')
+      )
+      ->addOption(
+        'distribution',
+        false,
+        InputOption::VALUE_OPTIONAL,
+        $this->trans('commands.generate.profile.options.distribution'),
+        false
+      );
+  }
+
+  /**
+   * ProfileCommand constructor.
+   */
+  public function __construct() {
+    $translater = new TranslatorManager();
+    $stringConverter = new StringConverter();
+    $renderer = new TwigRenderer($translater, $stringConverter);
+    $renderer->setSkeletonDirs([__DIR__ . '/../../templates/']);
+    $fileQueue = new FileQueue();
+    $generator = new SubProfileGenerator();
+    $generator->setRenderer($renderer);
+    $generator->setFileQueue($fileQueue);
+    $httpClient = new Client();
+    $appRoot = \Drupal::root();
+    $configurationManager = new ConfigurationManager();
+    $site = new Site($appRoot, $configurationManager);
+    $extensionManager = new Manager($site, $appRoot);
+    $validator = new Validator($extensionManager);
+    $this->excludedDependencies = [];
+    parent::__construct($extensionManager, $generator, $stringConverter, $validator, $appRoot, $site, $httpClient);
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  protected function interact(InputInterface $input, OutputInterface $output) {
+    $io = new DrupalStyle($input, $output);
+    $validators = $this->validator;
+
+    try {
+      // A profile is technically also a module, so we can use the same
+      // validator to check the name.
+      $profile = $input->getOption('profile') ? $validators->validateModuleName($input->getOption('profile')) : null;
+    } catch (\Exception $error) {
```

Comment:
Nit: The catch should be on its own line. And it's better to catch a more specific exception, if available, than just \Exception.